sql/core/src/test/resources/sql-tests/results/ansi/string-functions.sql.out

```diff
@@ -848,7 +849,8 @@ org.apache.spark.sql.AnalysisException
   "messageParameters" : {
     "actualNum" : "1",
     "expectedNum" : "2",
-    "functionName" : "`decode`"
+    "functionName" : "`decode`",
+    "sqlExpr" : "\"encode(abc, utf-8)\""
```

Comment:
The error message confuses slightly:
```
"Cannot resolve "encode(abc, utf-8)" due to data type mismatch:"
```
Actually, the issue is not in `encode` but in `decode`. `sqlExpr` should point out to the parent expression. @itholic Could you change this?

Reply:
Let me handle it. Thanks!

Reply:
ok. Ping me when you finish this.